packages/backend/src/nest/local-db/local-db.service.ts

```diff
@@ -344,4 +348,38 @@ export class LocalDbService {
       await this.put(key, arr)
     }
   }
+
+  public async addPendingQssSyncMessage(unsentMessage: QSSDataSyncMessage): Promise<void> {
+    const key = LocalDBKeys.PENDING_QSS_SYNCS
+    const arr: string[] = (await this.get(key)) || []
+    const messageString = JSON.stringify(unsentMessage)
+    if (arr.filter(pendingMessage => pendingMessage === messageString)) {
+      arr.push(messageString)
+      await this.put(key, arr)
+    }
+  }
+
+  public async getPendingQssSyncMessages(): Promise<QSSDataSyncMessage[]> {
+    const key = LocalDBKeys.PENDING_QSS_SYNCS
+    const arr: string[] = (await this.get(key)) || []
+    const parsed: QSSDataSyncMessage[] = []
+    for (const pendingMessage of arr) {
+      parsed.push(JSON.parse(pendingMessage) as QSSDataSyncMessage)
+    }
+    return parsed
+  }
+
+  public async removePendingQssSyncMessages(sentMessages: QSSDataSyncMessage[]): Promise<void> {
+    const key = LocalDBKeys.PENDING_QSS_SYNCS
+    let arr: string[] = (await this.get(key)) || []
+    for (const message of sentMessages) {
+      const messageString = JSON.stringify(message)
+      arr = arr.filter(pendingMessage => pendingMessage === messageString)
```

Comment:
```suggestion
      arr = arr.filter(pendingMessage => pendingMessage !== messageString)
```
I think the logic needs to be this to remove messages.